src/diffusers/models/cross_attention.py

```diff
@@ -185,17 +185,21 @@ def get_attention_scores(self, query, key, attention_mask=None):
             query = query.float()
             key = key.float()
 
+        if attention_mask is None:
+            baddbmm_input = torch.zeros(
+                query.shape[0], query.shape[1], key.shape[1], dtype=query.dtype, device=query.device
+            )
+        else:
+            baddbmm_input = attention_mask
+
         attention_scores = torch.baddbmm(
-            torch.empty(query.shape[0], query.shape[1], key.shape[1], dtype=query.dtype, device=query.device),
+            baddbmm_input,
             query,
             key.transpose(-1, -2),
-            beta=0,
+            beta=1,
```

Comment:
`beta` being 0 when `attention_mask is None` had a special meaning: it skips addition altogether. the input tensor is unused. this saves us any cost of allocation.

Reply:
ah, thanks!

Reply:
fixed here https://github.com/huggingface/diffusers/pull/2111/commits/ab1a42d05f1898acf3e4004164e5b234bbe3fa21